creator-node/src/routes/files.js

```diff
@@ -172,208 +167,106 @@ const getCID = async (req, res) => {
   const trackId = parseInt(req.query.trackId)
 
-  const decisionTree = [{ stage: `BEGIN`, time: `${Date.now()}` }]
-  const logPrefix = `[getCID] [CID=${CID}]`
+  const prometheusResult = { result: null, mode: 'default' }
 
   /**
    * Check if CID is servable from BlacklistManager; return error if not
    */
-  let startMs = Date.now()
   const BlacklistManager = req.app.get('blacklistManager')
   const isServable = await BlacklistManager.isServable(CID, trackId)
-  decisionTree.push({
-    stage: `BLACKLIST_MANAGER_CHECK_IS_SERVABLE`,
-    time: `${Date.now() - startMs}ms`
-  })
   if (!isServable) {
-    decisionTree.push({
-      stage: `CID_IS_BLACKLISTED`
-    })
-
-    logGetCIDDecisionTree(decisionTree, req)
-    return sendResponse(
+    prometheusResult.result = 'abort_delisted'
+    return sendResponseWithMetric(
       req,
       res,
-      errorResponseForbidden(
-        `${logPrefix} CID has been blacklisted by this node`
-      )
+      errorResponseForbidden('CID has been delisted by this node'),
+      prometheusResult
     )
   }
 
+  const storagePaths = []
+
   // Compute expected storagePath for CID
-  let storagePath
   try {
-    storagePath = await DiskManager.computeFilePath(CID, false)
-    decisionTree.push({
-      stage: `COMPUTE_FILE_PATH_COMPLETE`
+    const currentStoragePath = await DiskManager.computeFilePath(CID, false)
+    storagePaths.push({
+      path: currentStoragePath,
+      pathType: 'new_storage_path'
     })
-  } catch (e) {
-    decisionTree.push({
-      stage: `COMPUTE_FILE_PATH_FAILURE`
+
+    const legacyStoragePath = await DiskManager.computeLegacyFilePath(CID)
+    storagePaths.push({
+      path: legacyStoragePath,
+      pathType: 'legacy_storage_path'
     })
-    logGetCIDDecisionTree(decisionTree, req)
-    return sendResponse(
+  } catch (e) {
+    prometheusResult.result = 'abort_bad_cid'
+    return sendResponseWithMetric(
       req,
       res,
-      errorResponseBadRequest(`${logPrefix} Invalid CID`)
+      errorResponseBadRequest('Invalid CID'),
+      prometheusResult
     )
   }
 
   /**
-   * Check if file exists on FS at storagePath
+   * Check if file exists on FS at storagePath with the new and legacy storage paths
    * If found and not of file type, return error
    * If found and file type, continue
    * If not found, continue
    */
-  startMs = Date.now()
   let fileFoundOnFS = false
-  let fsStats
-  try {
-    /**
-     * fs.stat returns instance of fs.stats class, used to check if exists, an dir vs file
-     * Throws error if nothing found
-     * https://nodejs.org/api/fs.html#fspromisesstatpath-options
-     */
-    fsStats = await fs.stat(storagePath)
-    decisionTree.push({
-      stage: `FS_STATS`,
-      time: `${Date.now() - startMs}ms`
-    })
+  let storagePath, fsStats
 
-    if (fsStats.isFile()) {
-      decisionTree.push({
-        stage: `CID_CONFIRMED_FILE`
-      })
+  for (let i = 0; i < storagePaths.length; i++) {
```

Comment:
`for await (const storagePath of storagePaths) {}` better here right? you don't need the index

Reply:
or even better `for await const { path, pathType } of storagePaths) {}`